lib/matplotlib/ticker.py

```diff
@@ -1238,40 +1258,39 @@ def format_eng(self, num):
         u'-1.00 \N{GREEK SMALL LETTER MU}'
 
         `num` may be a numeric value or a string that can be converted
-        to a numeric value with the `decimal.Decimal` constructor.
+        to a numeric value with `float(num)`.
         """
-        dnum = decimal.Decimal(str(num))
-
+        dnum = float(num)
         sign = 1
 
         if dnum < 0:
             sign = -1
             dnum = -dnum
 
         if dnum != 0:
-            pow10 = decimal.Decimal(int(math.floor(dnum.log10() / 3) * 3))
+            pow10 = int(math.floor(math.log10(dnum) / 3) * 3)
         else:
-            pow10 = decimal.Decimal(0)
-
-        pow10 = pow10.min(max(self.ENG_PREFIXES))
-        pow10 = pow10.max(min(self.ENG_PREFIXES))
+            pow10 = 0
+            # Force dnum to zero, to avoid inconsistencies like
+            # format_eng(-0) = "0" and format_eng(0.0) = "0"
+            # but format_eng(-0.0) = "-0.0"
+            dnum = 0.0
+
+        pow10 = np.clip(pow10, min(self.ENG_PREFIXES), max(self.ENG_PREFIXES))
+
+        mant = sign * dnum / (10.0 ** pow10)
+        # Taking care of the cases like 999.9..., which
+        # may be rounded to 1000 instead of 1 k.
+        if (self.places is not None and
```

Comment:
I have another idea on how to handle this specific case, which would also be abe to take care of the "%g" format case. A dedicated commit is on its way.